src/test/java/com/epam/page/object/generator/integrationalTests/MainTest.java

```diff
@@ -172,4 +172,63 @@ public void pageObjectsGenerator_GenerateDropdownElementWithInnerElements() thro
         pog.generatePageObjects();
     }
 
+    @Test(expected = ValidationException.class)
+    public void pageObjectGenerator_notExistenceRootTitle() throws Exception {
+        PageObjectsGenerator pog = initPog(
+            "src/test/resources/dropdown-without-root.json",
+            "http://materializecss.com/dropdown.html",
+            false,
+            false);
+
+        pog.generatePageObjects();
+
+    }
+
+    @Test(expected = ValidationException.class)
+    public void pageObjectGenerator_duplicateInnerRoot() throws Exception {
+        PageObjectsGenerator pog = initPog(
+            "src/test/resources/dropdown-duplicate-root.json",
+            "http://materializecss.com/dropdown.html",
+            false,
+            false);
+
+        pog.generatePageObjects();
+
+    }
+
+    @Test(expected = ValidationException.class)
+    public void pageObjectGenerator_notExistenceTitleIntoInnerSearchRule() throws Exception {
+        PageObjectsGenerator pog = initPog(
+            "src/test/resources/dropdown-wrong-title.json",
+            "http://materializecss.com/dropdown.html",
+            false,
+            false);
+
+        pog.generatePageObjects();
+
+    }
+
+    @Test(expected = ValidationException.class)
+    public void pageObjectGenerator_notUniquenessAttributeIntoRoot() throws Exception {
+        PageObjectsGenerator pog = initPog(
+            "src/test/resources/dropdown-root-without-uniqueness.json",
+            "http://materializecss.com/dropdown.html",
+            false,
+            false);
+
+        pog.generatePageObjects();
+
+    }
+
+    @Test(expected = ValidationException.class)
+    public void pageObjectGenerator_DuplicateUniquenessAttributeIntoRule() throws Exception {
+        PageObjectsGenerator pog = initPog(
+            "src/test/resources/dropdown-duplicate-uniqueness.json",
+            "http://materializecss.com/dropdown.html",
+            false,
+            false);
+
+        pog.generatePageObjects();
+
+    }
```

Comment:
Need to add success cases.

Reply:
In this class we have three success test one for each type of SearcRle:
pageObjectsGenerator_success (commonSearchRule),
pageObjectGenerator_FormSuccess (formSearchRule),
pageObjectsGenerator_GenerateDropdownElementWithInnerElements (complexSearchRule).
Also in this sprint we have new team members and they started writing proper integration tests, so they will be rewrite.